bulldozer/merge.go

```diff
@@ -56,7 +57,15 @@ func MergePR(ctx context.Context, pullCtx pull.Context, client *github.Client, m
 			if err != nil {
 				return errors.Wrap(err, "failed to determine pull request body")
 			}
+
 			commitMessage = body
+			if opt.MessageDelimiter != "" {
```

Comment:
I think it might be worthwhile to refactor the calculation of `commitMessage` so that its more simple to write some tests around the expected behaviour. Perhaps refactoring something like L45-L79:
```
func calculateCommitMessage() (string) {
  commitMessage := ""
   if mergeConfig.Method == SquashAndMerge {

  return commitMessage
}
```

The only bit that would worry me is the inclusion of the `MustCompile` on L65 that will panic if the regex doesn't compile correctly, so that would be where I would want to focus on for testing. Maybe we can do that in another PR, or here.